crates/autopilot/src/infra/persistence/dto/quote.rs

```diff
@@ -1,18 +1,31 @@
 use {
     crate::{boundary, domain},
     number::conversions::big_decimal_to_u256,
+    primitive_types::U256,
 };
 
-pub fn into_domain(
-    quote: boundary::database::orders::Quote,
-) -> Result<domain::Quote, AmountOverflow> {
+pub fn into_domain(quote: boundary::database::orders::Quote) -> Result<domain::Quote, Error> {
     Ok(domain::Quote {
         order_uid: domain::OrderUid(quote.order_uid.0),
-        sell_amount: big_decimal_to_u256(&quote.sell_amount).ok_or(AmountOverflow)?,
-        buy_amount: big_decimal_to_u256(&quote.buy_amount).ok_or(AmountOverflow)?,
+        sell_amount: big_decimal_to_u256(&quote.sell_amount).ok_or(Error::AmountOverflow)?,
+        buy_amount: big_decimal_to_u256(&quote.buy_amount).ok_or(Error::AmountOverflow)?,
+        fee: {
+            let gas_amount = U256::from_f64_lossy(quote.gas_amount);
+            let gas_price = U256::from_f64_lossy(quote.gas_price);
+            let sell_token_price = U256::from_f64_lossy(quote.sell_token_price);
```

Comment:
nit: Not sure if the values are such that it would make a difference but intuitively I would say it should be more accurate and easier to read to do the computation using floats and only convert the final result to `U256`.

Reply:
Yeah, can do both. Working with f64 would silently lead to wrong values in case of overflow or division by zero, but don't expect this to happen ever in practice.